substrate/frame/system/src/lib.rs

```diff
@@ -1994,7 +2060,14 @@ impl<T: Config> Pallet<T> {
 	/// This needs to be used in prior calling [`initialize`](Self::initialize) for each new block
 	/// to clear events from previous block.
```

Comment:
It might be helpful to document that this doesn’t actually clear storage if event segments are in use.